lib/Doctrine/Common/DataFixtures/Purger/MongoDBPurger.php

```diff
@@ -67,7 +67,7 @@ public function purge()
         $metadatas = $this->dm->getMetadataFactory()->getAllMetadata();
         foreach ($metadatas as $metadata) {
             if ( ! $metadata->isMappedSuperclass) {
-                $this->dm->getDocumentCollection($metadata->name)->drop();
+                $this->dm->getDocumentCollection($metadata->name)->remove(array());
```

Comment:
If you don't need to preserve indexes, it's faster to simply drop the collection.